ttnn/cpp/ttnn/compiler_interface/compiler_interface.hpp

```diff
@@ -0,0 +1,31 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <tuple>
+
+#include "ttnn/compiler_interface/singleton_device_context.hpp"
+#include "ttnn/cpp/ttnn/tensor/enum_types.hpp"
+#include "ttnn/tensor/types.hpp"
+
+namespace ttnn::compiler_interface {
+
+using OperandParams =
```

Comment:
fyi, this pr https://github.com/tenstorrent/tt-metal/pull/15028 introduces TensorSpec. I think this is what you want here.